packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -58,7 +59,22 @@ const DefaultChunkSize = 16 * 1024;
 const ImmediateNoOpResponse = "";
 
 // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-const getRetryDelayFromError = (error: any): number | undefined => error?.retryAfterSeconds;
+export const getRetryDelayFromError = (error: any): number | undefined => error?.retryAfterSeconds;
+
+export function emitThrottlingWarning(
+    delayTime: number,
+    error: ICriticalContainerError,
+    emitter: Container | DeltaManager,
+) {
+    if (delayTime > 0) {
+        const throttlingError: IThrottlingWarning = {
+            errorType: ContainerErrorType.throttlingError,
+            message: `Service busy/throttled: ${error.message}`,
+            retryAfterSeconds: delayTime,
+        };
+        emitter.emit("throttled", throttlingError);
```

Comment:
In general, this is not good practice to raise events on some object. Better for this object to expose some API that we call into, and it raises events. Maybe other way to put it - EventEmitter should not be base class for our objects, we should instead hide it and use other means of composition, objects should only expose interfaces, like IDeltaManagerInternalEvents.

Reply:
Removed. Moved this logic to container.ts which will be common for delta and storage.